src/stores/AppStore/AppStore.ts

```diff
@@ -1928,6 +1932,10 @@ export class AppStore {
 
             for (let profile of spectralProfileData.profiles) {
                 profileStore.setProfile(ProtobufProcessing.ProcessSpectralProfile(profile, spectralProfileData.progress));
+                if (process.env.REACT_APP_SKIP_TELEMETRY === "false" && spectralProfileData.progress >= 1 && spectralProfileData.regionId) {
+                    const region = frame.getRegion(spectralProfileData.regionId);
+                    TelemetryService.Instance.addSpectralProfileEntry(region.regionType, region.regionId, region.size.x, region.size.y, frame.depthNumber);
```

Comment:
I would suggest we send one data with the info of profile number. Also, we need to exclude the cursor region.

Reply:
Isn't the cursor region already excluded?

Reply:
If it's excluded then it's fine. Sorry, I mean the length of `spectralProfileData.profiles`.